### DISCUSSION TITLE
HELP] How do I make presentations work with mobile browsers?

### DISCUSSION THREAD

**Discussion by hoshi411** [senshu/Sozi#614] [Q&A] [answered]
I am working on an app for analytical psychology research.
You can see my project here:
https://function-spaces.tiiny.site/

It works wonderfully on desktop browsers, but the buttons I have created do not work on mobile browsers.
Touch input events do not seem to work.

What is the problem?
Is there a way I can implement this myself?
Is there an easy way inside of Inkscape to implement this?

Not only do I want touch events so that the links will open as they do on the desktop,
I also want ontouch event to change the button color after touching so that it registers that it had been touched.
I also want onload event to show mobile users a quick button animation so that the user knows that there are buttons there.  This would just raise the opacity for a second and then lower it back.

What am I missing guys?

Thank you.

**Comment by aumouvantsillage** [comment]
Touch support was introduced in Sozi 21.08. I have not observed the problem myself but it seems that links no longer work.
There is no obvious solution at the moment, except using an earlier version such as Sozi 21.02.

> I also want ontouch event to [...]

These features can be implemented by creating a JavaScript file that you can add to your presentation. See the section "Custom stylesheets and scripts" in the right pane of the presentation editor.

**Comment by hoshi411** [comment]
Version: 22.02.03-1643876128
updated file: https://binary-personality.github.io/index.html

I don't think I explained correctly.
I am able to use touch to move through the slides on mobile but I am not able to use the html buttons that I have created.
If you open the site on desktop, you will see what I mean.
I made physical buttons which link to different areas of the slide.
(I was also crushed to see that using button links also kills the animations)
When on mobile we are unable to tap on the buttons and go to the correct slide.

I am able to tap with mobile and run through the slides in static order but that kills the whole project.

**Reply by aumouvantsillage** [reply]
> I am able to use touch to move through the slides on mobile but I am not able to use the html buttons that I have created.

Yes. Let me rephrase my previous answer: when touch support was introduced in Sozi 21.08, I think that it broke some other features that worked previously, such as tapping on links.

**Reply by hoshi411** [reply]
Thank you so much for taking time to reply. I love your software.  I am so excited about what sozi is allowing me to do right now.

Thank you for your patience with me.

I would like to try to continue using sozi in its current version but improve my svg somehow.
Would adding javascript into my svg fix the issue (ontouch events or somethin like that?)

**Comment by hoshi411** [comment, marked as answer]
I don't know why but it is working now perfectly.
I can use my mouse or finger to touch buttons and they change frames.
Here is what I suspect happened.

I possibly only uploaded the html file alone to the server without uploading the svg file.
I guess I began to assume from talks here that the html file was all I needed.

I realize that the SVG is linked so that I need to upload the html, json, and svg together.

.....
I don't know why but it works perfectly.  I don't need to add ontouch events.

that is amazing!

Thank you.

**Reply by hoshi411** [reply]
I also needed to turn off keyboard and mouse navigation.

**Comment by RW-F22** [comment]
Hi ! 
First thank you for this super tool !
I have the same problem : 
I included a link in a SVG drawing. It works with a computer but not on a touch screen.

Bonjour !
D'abord merci pour e super outil ! 
J'ai le même problème : 
le lien inclus dans le fichier SVG fonctionne sur un ordinateur, mais pas sur un écran tactile.

Does somebody find the reason why ?
Quelqu'un a-t-il trouvé l'explication ?

> I realize that the SVG is linked so that I need to upload the html, json, and svg together.

Are you confirming that?

> I also needed to turn off keyboard and mouse navigation.

Could you explain a little bit more ?
